### PR TITLE
3dsx: Fix per-game configuration

### PR DESCRIPTION
**Pull request by mtheall**
This fixes a handful of issues:
- Very slow framerate when running 3dsx, until causing Settings::Apply() to get called
  - Pressing Ok on either global or per-game config window
  - Pressing Ok on telemetry dialog
- Per-game configuration writes to $CONFIG/\<fullpath>/name.3dsx.ini
  - Changed to $CONFIG/name.3dsx.ini to be consistent with startup
- 3DSX loader doesn't implement ReadProgramId(), although code appears to assume 3dsx title id 0 in surrounding contexts and logs
  - This caused the initial per-game config loading in citra_qt/main.cpp to get skipped.
  
I went ahead and made the same changes for elf

- - -
This change is [Reviewable](https://reviewable.io/reviews/citra-emu/citra/6367)

### CODE REVIEW

**Comment by mtheall** [comment]
I'm not sure what's the root cause of the very slow framerate before Settings::Apply()

File: src/core/loader/3dsx.cpp [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// Copyright 2014 Citra Emulator Project
+// Copyright 2023 Citra Emulator Project
```
Comment (GPUCode) [inline src/core/loader/3dsx.cpp:1]: The license date is when the file was created, you don't need to change it. ditto for all other changes.

**Review by GPUCode — APPROVED** [review]

File: src/core/loader/3dsx.cpp [hunk 2]
```diff
@@ -328,6 +328,11 @@ ResultStatus AppLoader_THREEDSX::ReadRomFS(std::shared_ptr<FileSys::RomFSReader>
     return ResultStatus::ErrorNotUsed;
 }
 
+ResultStatus AppLoader_THREEDSX::ReadProgramId(u64& out_program_id) {
+    out_program_id = 0;
+    return ResultStatus::Success;
+}
+
```
Comment (BreadFish64) [inline src/core/loader/3dsx.cpp:335]: Hmm, do we want to implement these or just check for the unimplemented status code?

**Comment by FearlessTobi** [comment]
ping again, would be nice to get this merged soon

**Comment by endrift** [comment]
Would be nice to get this merged for homebrew testing.

**Comment by FearlessTobi** [comment]
Superseded by https://github.com/citra-emu/citra/pull/6691.

**Pull request closed without merging**